airflow/models/serialized_dag.py

```diff
@@ -114,14 +114,17 @@ def write_dag(cls, dag: DAG, min_update_interval: Optional[int] = None, session:
         # If Yes, does nothing
         # If No or the DAG does not exists, updates / writes Serialized DAG to DB
         if min_update_interval is not None:
-            if session.query(
-                exists().where(
+            if (
+                session.query(literal(True))
+                .filter(
                     and_(
                         cls.dag_id == dag.dag_id,
                         (timezone.utcnow() - timedelta(seconds=min_update_interval)) < cls.last_updated,
                     )
                 )
-            ).scalar():
+                .first()
+                is not None
```

Comment:
why this change, scalar() is not DB specific is it? or sqlalchmey does not convert it well too?

Reply:
the version of sqlalchemy which we use converts it wrongly for MySQL - https://github.com/sqlalchemy/sqlalchemy/issues/5481

Reply:
Could you add a todo comment here to change this back once we are on SQL 1.4-- looks like the fixed the issue.?

Reply:
done